### ISSUE TITLE
[MJAVADOC-179] Aggregate report ignores sourcepath customization of subprojects

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#954]
**[Fol De Rol](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=folderol)** opened **[MJAVADOC-179](https://issues.apache.org/jira/browse/MJAVADOC-179?redirect=false)** and commented

When using aggregate javadoc reporting with one of subproject being customized to include java sources from alternative directories, only original source directories are passed to javadoc executable.

For instance, I use POJO's generated by Hibernate3 from HBM files which are documented. The generated files are placed into ${basedir}/target/hibernate3/generated-sources/ directory. My subproject's POM contains the following definition:

```xml
<reporting>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <version>2.2</version>
                <configuration>
                  <sourcepath>$\{basedir\}/src/main/java;$\{basedir]}/target/hibernate3/generated-sources</sourcepath>
                </configuration>
            </plugin>
        </plugins>
    </reporting>
```

and it works correctly if running `mvn site` on subproject level. If run from the parent project, as I traced, the custom directory does not passed to the javadoc tool.

---

**Affects:** 2.3

**Issue Links:**
- [MJAVADOC-181](https://issues.apache.org/jira/browse/MJAVADOC-181) Javadoc report not generated for multi-module project if run from parent level.

- [MJAVADOC-168](https://issues.apache.org/jira/browse/MJAVADOC-168) Regression: 2.4-SNAPSHOT does not generate docs for generates sources if run outside a build

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

I cant reproduce it with 2.4 and 2.5-SNAPSHOT. So, try with these versions and feel free to reopen it if the pb is still there (Think to including a test case...)
Thanks.

**Comment by jira-importer** [comment]
**[Fol De Rol](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=folderol)** commented

Still reproduced in 2.4

**Comment by jira-importer** [comment]
**[Fol De Rol](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=folderol)** commented

A test case:

fragment of the parent POM:

    <reporting>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <version>2.4</version>
                <configuration>
                    <aggregate>true</aggregate>
                </configuration>
            </plugin>
        </plugins>
    </reporting>

fragment of the sub-module POM:

    <reporting>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <version>2.4</version>
                <configuration>
                    <sourcepath>${basedir}/src/main/java;${basedir}/target/hibernate3/generated-sources</sourcepath>
                </configuration>
            </plugin>
        </plugins>
    </reporting>

not any class from target/hibernate3/generated-sources is aggregated into the complete javadoc.

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Should be fixed in r661899. Could you confirm that you used this revision?

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Without any feedbacks, we can't help you.

**Comment by jira-importer** [comment]
**[Fol De Rol](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=folderol)** commented

I confirm the issue is fixed in 2.5.
